dvc/repo/experiments/show.py

```diff
@@ -177,26 +177,29 @@ def _build_rows(
                 )
 
 
-def _sort_column(
+def _sort_column(  # noqa: C901
     sort_by: str,
     metric_names: Mapping[str, Iterable[str]],
     param_names: Mapping[str, Iterable[str]],
 ) -> Tuple[str, str, str]:
-    path, _, sort_name = sort_by.rpartition(":")
+    sep = ":"
+    parts = sort_by.split(sep)
     matches: Set[Tuple[str, str, str]] = set()
 
-    if path:
+    for split_num in range(1, len(parts)):
+        path = sep.join(parts[:split_num])
+        sort_name = sep.join(parts[split_num:])
         if path in metric_names and sort_name in metric_names[path]:
             matches.add((path, sort_name, "metrics"))
         if path in param_names and sort_name in param_names[path]:
             matches.add((path, sort_name, "params"))
```

Comment:
I think this would have to go from `range(0, len(parts))`, it has to cover the case where there is no path, and only a metric containing a colon

IMO this is also more confusing than requiring escaping, and I think there are still edge cases where the user could have filenames and metric names that overlap and break this kind of behavior

Reply:
> I think this would have to go from `range(0, len(parts))`, it has to cover the case where there is no path, and only a metric containing a colon

Thanks, fixed.

> IMO this is also more confusing than requiring escaping, and I think there are still edge cases where the user could have filenames and metric names that overlap and break this kind of behavior

My thoughts on this PR:
1. It requires nothing of the user
2. Edge cases will show the same error about being ambiguous that shows now unless I miss something
3. Managing escape characters between how they get read in the user's shell and how we handle them seems more confusing to me (and more for the user to understand)
4. Don't see much harm since the implementation already exists

Do you feel it's a blocker and escaping is the only way to go?

Reply:
Given that the existing behavior is already broken, we can merge this.

But my objection is more that this same problem affects every other command/flag combination that uses `path:metric_or_param` addressing (like `exp run --set-param` and probably also `stage add --params`). I would prefer to only have to fix this once, in a consistent and re-usable way. This PR is just patching over one symptom of a broader underlying problem

Reply:
Also in the same vein, I'm not sure that we actually document that `.` is an invalid character in metric or parameter names, but for all intents and purposes we don't support it because we treat `.` as the dictionary level separator in metric/param addressing (even though it is valid to have string keys that contain `.` in yaml/json). It would be easier for us to just treat `:` the same way and disallow it entirely in metric/param names

Reply:
Valid points. As mentioned in the issue description, I don't see much harm in doing this unless/until we forbid these characters or have a better solution like escaping, so I'm going to merge but open an issue to discuss whether we should allow it.